### DISCUSSION TITLE
3.0.0rc1

### DISCUSSION THREAD

**Discussion by michael-s-molina** [apache/superset#24581] [General]
Hello Superset Community,

This is a call for the vote to release Apache Superset version 3.0.0.

The release candidate:
https://dist.apache.org/repos/dist/dev/superset/3.0.0rc1/

Git tag for the release:
https://github.com/apache/superset/tree/3.0.0rc1

The Change Log for the release:
https://github.com/apache/superset/blob/3.0.0rc1/CHANGELOG.md

The Updating instructions for the release:
https://github.com/apache/superset/blob/3.0.0rc1/UPDATING.md

Public keys are available at:
https://www.apache.org/dist/superset/KEYS

The vote will be open for at least 72 hours or until the necessary number
of votes are reached.

Please vote accordingly:

[ ] +1 approve
[ ] +0 no opinion
[ ] -1 disapprove with the reason

Thanks,
The Apache Superset Team

<hr /><em>This discussion was created from the release <a href='https://github.com/apache/superset/releases/tag/3.0.0rc1'>3.0.0rc1</a>.</em>

**Comment by padbk** [comment]
Great to see this release. Thank you!

We had a bug on a migration, where there was a dashboard with a deleted chart on it. In a row in the DB: `dashboards` table, `json_metadata` column, `chart_configuration` key, had the value `{"undefined": true}`. This caused the following line to error  https://github.com/apache/superset/blob/eb42fa7a2e21d5d4d69a6e8a9787dc8ea1dd6348/superset/migrations/versions/2023-05-11_12-41_4ea966691069_cross_filter_global_scoping.py#L65

I modified the value of that key in the DB  to `{}` and the migration completed fine. Just flagging this in case you want to code for this in the migration.

**Reply by michael-s-molina** [reply]
@kgabryje will look into it 🙏🏼

**Reply by kgabryje** [reply]
Hey @padbk, thanks for reporting that! Here's a PR that addresses this bug: https://github.com/apache/superset/pull/24661

**Comment by mattitoo** [comment]
I tested locally with 3.0.0rc1 via docker.
Clicking on a tag in the dashboard and charts lists does not work, it links to http://localhost:8088/superset/all_entities/?tags=Test which leads to a 404 error. 
Anyone else have this issue?

**Reply by padbk** [reply]
Yes, same 404. But the link is https://oursuperseturl.com/superset/all_entities/?tags=Test for us

**Reply by michael-s-molina** [reply]
Fix: https://github.com/apache/superset/pull/24593

**Comment by alx25** [comment]
I'm testing 3.0.0rc1, and I have a question regarding the use of HTML in tables. In 2.1.0, images display correctly, but in 3.0.0rc1, only the link is shown without the actual image. Will this be a permanent feature in the final 3.0.0 version, or could it be a bug?

In 2.1.0:
![image](https://github.com/apache/superset/assets/50088645/e8d92d66-e866-48c5-9298-ee8c869fb974)

Now in 3.0.0rc1:
![image](https://github.com/apache/superset/assets/50088645/9d7a86b2-913c-4355-bc94-825fe35abddb)

**Reply by michael-s-molina** [reply]
This may be related to https://github.com/apache/superset/pull/24262. Could you check the Network and Console tabs in Chrome Dev Tools for errors?

**Reply by alx25** [reply]
I see, you're right. The issue is related to CSP

![image](https://github.com/apache/superset/assets/50088645/a8d91ef0-3224-4e37-a7f0-2a9f83b490f2)

![image](https://github.com/apache/superset/assets/50088645/87f3ad53-27e6-4ffa-b51c-1935ef3674a1)

**Reply by michael-s-molina** [reply]
👍🏼

**Reply by michael-s-molina** [reply]
You may wanna check the CSP section [here](https://github.com/apache/superset/blob/3.0/docs/docs/security.mdx) to fix these errors in your deployment.

**Comment by michael-s-molina** [comment]
Incorrect filter scope when using tabs https://github.com/apache/superset/issues/24595

**Reply by michael-s-molina** [reply]
Fix: https://github.com/apache/superset/pull/24608

**Comment by mdeshmu** [comment]
We are not able to login using oauth in 3.0.0.rc1. It is already reported in https://github.com/apache/superset/issues/24597

**Reply by michael-s-molina** [reply]
Thank you for adding the issue reference.

**Reply by mdeshmu** [reply]
We can't even test further as we cannot login :)
Any workaround would be appreciated until a permanent fix is implemented.

**Reply by michael-s-molina** [reply]
You can change your configurations and disable `TALISMAN_ENABLED` or set `content_security_policy` to `None` in `TALISMAN_CONFIG`.

**Reply by michael-s-molina** [reply]
Fix: https://github.com/apache/superset/pull/24646

**Comment by michael-s-molina** [comment]
Time Grain issue on pivot-table-v2: https://github.com/apache/superset/issues/24460

**Reply by michael-s-molina** [reply]
Fix: https://github.com/apache/superset/pull/24665

**Reply by michael-s-molina** [reply]
Fix for similar bug on Table chart: https://github.com/apache/superset/pull/24746

**Comment by padbk** [comment]
Seeing a bug with csv or text alerts on 3.0.0rc1, which were working fine on 2.1.0. 
![image](https://github.com/apache/superset/assets/29596014/58146b12-376a-4c83-b345-0aed7a117615)
PNG is working fine on 3.0.0rc1 (which rules out auth/selenium issues), but CSV and text are failing with:
```
A downstream exception occurred while generating a report: b056bed4-ff76-4e1d-aa0f-0fa7751d3542. Failed generating csv HTTP Error 400: BAD REQUEST
Traceback (most recent call last):
  File "/app/superset/reports/commands/execute.py", line 262, in _get_csv_data
    csv_data = get_chart_csv_data(chart_url=url, auth_cookies=auth_cookies)
  File "/app/superset/utils/csv.py", line 91, in get_chart_csv_data
    response = opener.open(chart_url)
  File "/usr/local/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/usr/local/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/usr/local/lib/python3.9/urllib/request.py", line 561, in error
    return self._call_chain(*args)
  File "/usr/local/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python3.9/urllib/request.py", line 641, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: BAD REQUEST
```

**Reply by michael-s-molina** [reply]
@eschutho @hughhhh Could you investigate this?

**Reply by john-bodley** [reply]
@michael-s-molina I can take a look at this issue.

**Reply by john-bodley** [reply]
Fix: https://github.com/apache/superset/pull/24737

**Comment by padbk** [comment]
I'm also investigating a bug where Thumbnails are computed every time and the cache is not checked. Looks like the `force` tag is defaulting to `true`.

**Reply by michael-s-molina** [reply]
Thank you @padbk. If you are able to reproduce the bug, please open an issue with the details and please reference it here.

**Reply by padbk** [reply]
Yeh, I've had to turn thumbnails off to avoid overloading the workers. I will try to find time to spin up another instance to debug if I can.

**Comment by jrog24** [comment]
Hello, I'm not able to login to UI of Superset 3.0. I've created a admin account, ran db upgrade and superset init, and when I'm trying to log into Superset, after clicking on "Sign in" nothing happens. Not a single error, just looping back to the login page. Tested on both Firefox and Chrome. Hosted on Docker (apache/superset:3.0.0rc1) on Debian 11.

**Reply by michael-s-molina** [reply]
Hi @jrog24. I think it's the same problem as reported [here](https://github.com/apache/superset/discussions/24581#discussioncomment-6373172).

**Reply by jrog24** [reply]
@michael-s-molina thank you very much, setting `TALISMAN_ENABLED` to `false` in config fixed my issue

**Comment by michael-s-molina** [comment]
Unable to run Superset 3.0.0rc1 on Debian https://github.com/apache/superset/issues/24638

**Reply by michael-s-molina** [reply]
@EugeneTorap will take a look at it.

**Reply by michael-s-molina** [reply]
Resolved without any fix.

**Comment by michael-s-molina** [comment]
Cache warmup only works for legacy chart types.

**Reply by michael-s-molina** [reply]
Fix: https://github.com/apache/superset/pull/24671

**Comment by ghost** [comment]
@michael-s-molina Facing this issue #24684 
Unable to connect to PSSQL database using RDS proxy endpoint

**Reply by michael-s-molina** [reply]
@betodealmeida is working on a fix in https://github.com/apache/superset/pull/24739 but this is not considered a blocker for the release. I'll cherry pick it if it gets fixed in time. If not, it will probably be cherry picked in a patch release.

**Reply by mdeshmu** [reply]
https://github.com/apache/superset/pull/24739 does fix the issue. Please include it in RC2.

**Comment by john-bodley** [comment]
@michael-s-molina we're facing issue https://github.com/apache/superset/issues/24729.

**Reply by michael-s-molina** [reply]
@justinpark will take a look at it.

**Reply by justinpark** [reply]
Fix: #24749

**Comment by michael-s-molina** [comment]
Select `onChange` is being fired without explicit selection.

**Reply by michael-s-molina** [reply]
Fix: https://github.com/apache/superset/pull/24698

**Comment by michael-s-molina** [comment]
Rich tooltip no longer highlights hovered-over data series in stacked barchart: https://github.com/apache/superset/issues/24738

**Reply by michael-s-molina** [reply]
Fix: https://github.com/apache/superset/pull/24756

**Comment by mdeshmu** [comment]
Please help with https://github.com/apache/superset/issues/24782

**Reply by michael-s-molina** [reply]
@john-bodley PR partially addresses this: https://github.com/apache/superset/pull/24789

**Reply by michael-s-molina** [reply]
This will be included in a next RC or patch release.

**Comment by mdeshmu** [comment]
Please address https://github.com/apache/superset/issues/24790

**Reply by michael-s-molina** [reply]
This will be included in a next RC or patch release.

**Comment by yolandazcc** [comment]
Please help with https://github.com/apache/superset/issues/23735

**Reply by michael-s-molina** [reply]
Fix: https://github.com/apache/superset/pull/24477

**Comment by sqja** [comment]
Hello,
Would you have any news regarding this regression https://github.com/apache/superset/issues/21508 which was spotted in 2.1.0 please?

**Reply by michael-s-molina** [reply]
@sqja Are you able to reproduce this on 3.0? You can test RC1 using [PyPi](https://pypi.org/project/apache-superset/3.0.0rc1/) or [Docker](https://hub.docker.com/layers/apache/superset/3.0.0rc1/images/sha256-350dcd494015de2a087e51751303b82e61cd7f8ea6c83cf03e2da456e4409de3?context=explore).

**Reply by michael-s-molina** [reply]
This will be included in a next RC or patch release.

**Comment by michael-s-molina** [comment]
Closing the discussion as [RC2](https://github.com/apache/superset/discussions/24824) is out.
